File: src/types/project.ts
export type ProductionPhase =
  | "idea"
  | "arrangement"
  | "sound-design"
  | "mixing"
  | "mastering"
  | "done";

export interface Project {
  id: number | null;
  name: string;
  bpm: number | null;
  key: string | null;
  genre: string | null;
  phase: ProductionPhase;
  notes: string | null;
  created_at: string | null;
  updated_at: string | null;
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Project {
    pub id: Option<i64>,
    pub name: String,
    pub bpm: Option<i32>,
    pub key: Option<String>,
    pub genre: Option<String>,
    pub phase: String,
    pub notes: Option<String>,
    pub created_at: Option<String>,
    pub updated_at: Option<String>,
}
